aip/general/0129.md

```diff
@@ -0,0 +1,71 @@
+---
+id: 129
+state: draft
+created: 2023-06-21
+placement:
+  category: resource-design
+  order: 65
+---
+
+# Server-modified Values and Defaults
+
+### State-driven clients (aka Convergent clients)
+Imperative clients rely on the result of an RPC call to determine success or failure of an operation. State-driven clients (including declarative clients) are clients that use the state of an existing resource to determine actions that should be taken to bring that resource to a desired state. To determine if a set of updates was successful a state-driven client compares server side resource attributes with client defined values. The comparison feature ensures accuracy of a creation or an update but it requires services to treat the client set fields as read-only and diligently preserve those values.
+
+### Permadiff
+Services that update client-set fields communicate to state-driven clients that the desired change has not taken effect, which results in the state-driven client to continuously call the service with updates. Permadiff is an infinite loop scenario where a client continuously attempts to update a resource field or a set of fields that the API service also modifies.
+
+## Guidance
+
+### Server Side Defaults
+There are instances where a service has to ensure that a value always exists even if the client chooses not to specify one. For example: a client may create a caching service instance without specifying the software version to be used for the service. Such a scenario requires the service to pick a version. The service designer has a number of ways to handle server side defaults.
```

Comment:
I think this is about hosted open source, such as Redis release 7.0.11.
https://redis.io/download/

Reply:
I suggest that's either clarified, or a different example is picked - readers shouldn't have to be joining dots.

Reply:
GCE has lots of examples, such as the default machine type.

Reply:
It's also a bit of a classic anti-example. Exactly the case where you should use two fields instead of one. 

Better would be an example where the default must not evolve for the lifetime of the field. Like a timeout, or a non-enum select-string like Sku: "Basic"